src/Runner.Common/JobServer.cs

```diff
@@ -32,27 +35,40 @@ public sealed class JobServer : RunnerService, IJobServer
         private VssConnection _connection;
         private TaskHttpClient _taskClient;
 
-        public async Task ConnectAsync(VssConnection jobConnection)
+        public async Task ConnectAsync(Uri jobServerUrl, VssCredentials jobServerCredential, DelegatingHandler[] delegatingHandler = null)
         {
-            _connection = jobConnection;
+            Trace.Info($"Establishing connection for JobServer");
             int attemptCount = 5;
-            while (!_connection.HasAuthenticated && attemptCount-- > 0)
+
+            while (attemptCount-- > 0)
             {
                 try
                 {
-                    await _connection.ConnectAsync();
+                    await RefreshConnectionAsync(jobServerUrl, jobServerCredential, delegatingHandler);
                     break;
                 }
                 catch (Exception ex) when (attemptCount > 0)
                 {
-                    Trace.Info($"Catch exception during connect. {attemptCount} attemp left.");
+                    Trace.Info($"Catch exception during connect. {attemptCount} attempts left.");
                     Trace.Error(ex);
                 }
 
                 await Task.Delay(100);
             }
 
             _taskClient = _connection.GetClient<TaskHttpClient>();
+        }
+
+        private async Task RefreshConnectionAsync(Uri jobServerUrl, VssCredentials jobServerCredential, DelegatingHandler[] delegatingHandler)
+        {
+            Trace.Info($"Refresh JobServer VssConnection to get on a different AFD node.");
+
+            VssConnection newConnection = null;
+
+            _hasConnection = false;
+            newConnection = VssUtil.CreateConnection(jobServerUrl, jobServerCredential, delegatingHandler);
+            await newConnection.ConnectAsync();
+            _connection = newConnection;
```

Comment:
do we need to `_connection?.Dispose()` before overwrite it?

Reply:
Good catch. I'll add it back.